tfjs-core/src/ops/array_ops.ts

```diff
@@ -26,6 +26,72 @@ import {op} from './operation';
 import {MPRandGauss, RandGamma, UniformRandom} from './rand';
 import {zeros, zerosLike} from './tensor_ops';
 
+/** Broadcast an array to a compatible shape NumPy-style.
+ *  
+ *  The tensor's shape is compared to the broadcast shape from end to beginning.
+ *  Ones are prepended to the tensor's shape until is has the same length as
+ *  the broadcast shape. If input.shape[i]==shape[i], they (i+1)-th axis is
+ *  already broadcast-compatible. If input.shape[i]==1 and shape[i]==N, then
+ *  the input tensor is tiled N times along that axis (using tf.tile).
+ *  
+ *  @param input The tensor that is to be broadcasted.
+ *  @param shape The input is to be broadcast to this shape.
+ */
+/** @doc {heading: 'Tensors', subheading: 'Transformations'} */
+function broadcastTo_<R extends Rank>(
+  x: Tensor|TensorLike, shape: ShapeMap[R]
+): Tensor<R>
+{
+  let input = convertToTensor(x, 'broadcastTo', 'x');
+  const xShape = input.shape;
+
+  if( shape.some(d => !(d > 0) || d%1 !== 0) ) {
+    throw new Error(`broadcastTo(): Invalid broadcast shape [${shape}].`);
+  }
+
+  if( shape.length < input.rank ) {
+    throw new Error(
+      `broadcastTo(): shape.length=${shape.length} < input.rank=${input.rank}.`
+    );
+  }
+
+  if( shape.length > input.rank )
+  {
+    const newShape =  input.shape.slice();
+    while(newShape.length < shape.length ) {
+          newShape.unshift(1);
+    }
+    input = input.reshape(newShape);
+  }
+
+  const reps: number[] = Array.from(shape);
+  for( let i=shape.length; i-- > 0; )
+  {
+    if( input.shape[i] === shape[i] ) {
+      reps[i] = 1;
+    }
+    else if( input.shape[i] !== 1 ) {
+      throw new Error(
+        `broadcastTo(): [${xShape}] cannot be broadcast to [${shape}].`
+      );
+    }
+  }
+
+  const axes = reps.map( ( n,i)  => n > 1 ? i : -1 ).filter( i => i >= 0 );
+
+  if( axes.length === 0 ) {
+    return input as Tensor<R>;
```

Comment:
Please clone the result before returning it, like this: `input.clone()`.

Reply:
Without `.clone()`, disposing the input would also dispose the return value, right? Didn't even think about that.